frontends/.eslintrc.js

```diff
@@ -79,6 +79,27 @@ module.exports = {
       },
     ],
     quotes: ["error", "double", { avoidEscape: true }],
+    "no-restricted-syntax": [
```

Comment:
TLDR: I added a few ESlint rules to encourage engineers to use named font styles (e.g., `theme.typography.body1`) rather than specifying weight and font-family.

The reasons are:
1. Our "design system" has a few named styles (`H1...H5, `Subtitle1...Subtitle4`, `body1...body4`, `buttonLarge, button, buttonSmall`). The designs should always be using these and we should be using them in the code, too. This helps with consistency and if we need to tweak one down the road, it will be easy.
2. But also... **Figma CSS is incorrect for font weight**.
    - For example, our subtitle font is is "Neue Haas Grotesk Display  65 Medium".
        - Adobe says this is `font-family: neue-haas-grotesk-display`, `font-weight: 600`. (See screenshot below).
        - Figma thinks this is `font-family: Neue Haas Grotesk Display Pro`, `font-weight: 500`. (See screenshot below).
    - The correspondence of css font descriptors (weight, style) to font glyphs is provided by a stylesheet. The same font glyphs can be called `font-weight: 500` by one stylesheet and `font-weight: 600` by another stylesheet. Figma does not have any way, as far as I can tell, to establish the `weight <--> font` correspondence. It seems to just guess based on heuristics like "semi-bold = 600, bold = 700, heavy = 800, black = 900`, etc.
        - this seems like a common issue with figma, e.g., https://forum.figma.com/t/how-should-we-communicate-web-font-css-specs-when-inspect-mode-is-wrong/19355 
    - Adobe provides the font stylesheet, so Adobe's values are the one we should trust. [^1]

**Adobe:**
<img width="978" alt="Screenshot 2024-05-13 at 1 19 08 PM" src="https://github.com/mitodl/mit-open/assets/9010790/456cf734-40e8-4cd2-ade3-da601df2f69f">

**Figma:**
<img width="200" alt="Screenshot 2024-05-13 at 1 18 47 PM" src="https://github.com/mitodl/mit-open/assets/9010790/2c45e032-860f-4890-80c0-1be32aa31b39"> <img width="347" alt="Screenshot 2024-05-13 at 1 18 36 PM" src="https://github.com/mitodl/mit-open/assets/9010790/1cbc2c96-2b62-4466-a964-6e266e5ea1d4">




[^1]: The adobe-provided stylesheet is https://use.typekit.net/lbk1xay.css. It includes a bunch of `@font-face` declarations and URLS to the fonts. I guess we could manually edit https://use.typekit.net/lbk1xay.css, but I don't think we should, since the format of the URLs in https://use.typekit.net/lbk1xay.css could change. Whereas the stylesheet https://use.typekit.net/lbk1xay.css itself is clearly public and intended for use.